client/src/components/PrivateRoute.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { Navigate } from 'react-router-dom';
+
+interface PrivateRouteProps {
+  children: React.ReactNode;
+  useDummyAuth?: boolean;
+}
+
+class DummyAuth {
+  private isAuthenticated: boolean;
+
+  constructor(auth: boolean) {
+    this.isAuthenticated = auth;
+  }
+
+  checkAuth() {
+    return this.isAuthenticated;
+  }
+}
+
+const PrivateRoute: React.FC<PrivateRouteProps> = ({ children, useDummyAuth = false }) => {
```

Comment:
this line is too long; you might want to insert newline after the first equals symbol.